crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/ProceedingService.java

```diff
@@ -69,6 +69,30 @@ public ApiUpdateApplicationResponse updateApplication(CrownCourtDTO dto) {
         return apiUpdateApplicationResponse;
     }
 
+
+    public Optional<Void> checkCCDetails(CrownCourtDTO dto) {
```

Comment:
It looks like this method only ever returns an empty `Optional` or throws an exception (it never returns an `Optional` with any contents). This looks like an unusual way to use an `Optional`, is there a reason this method cannot be coid? i.e.  `public void checkCCDetails(`

Reply:
We are using same validation pattern for most of microservice and easy way to test the validation method with Optional<void>